### PR TITLE
github-runner: 2.320.0 -> 2.321.0

### PR DESCRIPTION
**Pull request by YorikSar**
* remove mentions of Node.js 16 which is not supported anymore see https://github.com/actions/runner/pull/3503
* switch to .Net SDK 8 see https://github.com/actions/runner/pull/3500
* fix source path discovery in deterministic build: runtime assumes that the root of the project is `/_`, replace it with actual `/build/src` so that tests can find their data.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by YorikSar** [comment]
@NixOS/nix-formatting Why did nixfmt check [forced](https://github.com/NixOS/nixpkgs/actions/runs/11864577138/job/33068256231) me to make this change: https://github.com/NixOS/nixpkgs/compare/23b17df5a709373bd3cfc297994ea54d06506707..19ab04a20aade9f1ae21726c8f4304dff2cc47d5 even though I didn't touch those lines or anything around them? Did some formatting style change since last change to this file? If it did, why is it considered formatted before my commit?

File: pkgs/by-name/gi/github-runner/package.nix [hunk 1]
```diff
@@ -80,19 +80,16 @@ buildDotnetModule rec {
...
-                  'true'
-    ''
-    + lib.optionalString (nodeRuntimes == [ "node20" ]) ''
-      substituteInPlace src/Runner.Common/Util/NodeUtil.cs \
-        --replace-fail '_defaultNodeVersion = "node16"' '_defaultNodeVersion = "node20"'
-    '';
+  postPatch = ''
+    # Ignore changes to src/Runner.Sdk/BuildConstants.cs
+    substituteInPlace src/dir.proj \
+      --replace 'git update-index --assume-unchanged ./Runner.Sdk/BuildConstants.cs' \
```
Comment (drupol) [inline pkgs/by-name/gi/github-runner/package.nix:90]:
```suggestion
      --replace-fail 'git update-index --assume-unchanged ./Runner.Sdk/BuildConstants.cs' \
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gi/github-runner/package.nix (hunk 1 above)
Comment (YorikSar) [reply pkgs/by-name/gi/github-runner/package.nix:90]: I didn't actually change these lines, and there are plenty places in this file with `--replace`, but ok, I fixed this one instance.

**Review by drupol — APPROVED** [review]

**Review by kfollesdal — APPROVED** [review]

**Comment by kfollesdal** [comment]
Result of `nixpkgs-review pr 356300` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>github-runner</li>
  </ul>
</details>

**Pull request merged**